pkg/apis/batch/validation/validation.go

```diff
@@ -425,6 +548,26 @@ func validateJobStatus(status *batch.JobStatus, fldPath *field.Path) field.Error
 			}
 		}
 	}
```

Comment:
What is checking
1. successcriteriamet condition cannot be set if the .spec.successpolicy is not specified
2. complete cannot be set for a .spec.successpolicy job unless the successcriteriamet is set
3. successcriteriamet cannot change from true to anything else

Reply:
Thank you for pointing this out.
Regarding the third validation, we already verify it here: https://github.com/kubernetes/kubernetes/blob/241dbdebaf0b4a0c0829e6ffc49ada94313c461a/pkg/apis/batch/validation/validation.go#L556-L565

Please let me know if I'm missing anything. Thanks.

Reply:
I also don't see 3 validated.

What you are pointing out is that SuccessCriteriaMet cannot be set along with Failed or FailureTarget.

But nothing is preventing SuccessCriteriaMet from transitioning from True to False.

Reply:
> But nothing is preventing SuccessCriteriaMet from transitioning from True to False.

Ah, it makes sense. I should address the third validation as well. Thanks.

Reply:
Done.